### PR TITLE
1705BEE Eddie Rose 

### PR DESCRIPTION
**Pull request by erose357**
@mikedao @laurenfazah

### CODE REVIEW

File: spec/features/tags/user_sees_tags_associated_with_job_spec.rb [hunk 1]
```diff
@@ -3,10 +3,10 @@
 describe "User visits a job page" do
   scenario "also sees tags associated with that job" do
     company = Company.create(name: "Pizza Planet")
-    job = Job.create(title: "Cook", level_of_interest: 50, city: "Mars", salary: 35000)
-    tag = tag.create(name: "Food")
+    job = Job.create(title: "Cook", level_of_interest: 50, city: "Mars", salary: 35000, company_id: company.id)
+    tag = Tag.create(name: "Food")
```
Comment (laurenfazah) [inline spec/features/tags/user_sees_tags_associated_with_job_spec.rb]: Good catch!

File: app/views/jobs/show.html.erb [hunk 2]
```diff
@@ -11,3 +11,8 @@ City: <%= @job.city %>
 <br>
 
 Company: <%= link_to @job.company.name, company_path(@job.company) %>
+<br>
+
+<% @job.tags.each do |tag| %>
+  Tags: <%= tag.name %>
```
Comment (laurenfazah) [inline app/views/jobs/show.html.erb]: Great use of AR association methods

File: spec/features/tags/user_sees_tags_associated_with_job_spec.rb [hunk 3]
```diff
@@ -7,10 +7,28 @@
...
+    job2.tags << tag
+
+
+    visit company_job_path(company.id, job.id)
+
+    expect(page).to have_content(company.name)
+    expect(page).to have_content(job.title)
+    expect(page).to have_content(job.city)
+    expect(page).to have_content("Might be interesting - 2")
+  end
```
Comment (laurenfazah) [inline spec/features/tags/user_sees_tags_associated_with_job_spec.rb:33]: Great job building upon scenario with feature tests

File: app/models/tag.rb [hunk 4]
```diff
@@ -1,4 +1,9 @@
 class Tag < ActiveRecord::Base
   has_many :job_tags
   has_many :jobs, through: :job_tags
+
+
+  def average_salary
+    jobs.pluck(:salary).sum/jobs.count
+  end
```
Comment (laurenfazah) [inline app/models/tag.rb]: Well-optimized way of calculating this

Comment (laurenfazah) [reply app/models/tag.rb]: And then a great refactor!

File: spec/models/tag_spec.rb [hunk 5]
```diff
@@ -1,5 +1,20 @@
 require 'rails_helper'
 
-RSpec.describe Tag, type: :model do
-  pending "add some examples to (or delete) #{__FILE__}"
+describe Tag do
```
Comment (laurenfazah) [inline spec/models/tag_spec.rb]: Bonus validations - love it!

**Comment by laurenfazah** [comment]
Commits showed really clear and thoughtful process for coming to solutions. Feature tests really drove development and kept you on a strong path. Use of ActiveRecord was awesome - really loved seeing the approach you took and refactors you came back to make.

One thing worth spending more time on would be sad-path and edge-case testing. Tests to make sure things that should go wrong do go wrong and are handled well always strengthen your codebase.

Overall, well done!
